### PR TITLE
[dbus] Use default meson values when options are None

### PR DESCRIPTION
**Pull request by uilianries**
### Summary
Changes to recipe:  **dbus/1.15.8**

#### Motivation

This bug was reported by @garethsb. Thank you! 

Using option value as `None` is converted to `"None"`, resulting in misconfiguration when consuming the project.

Plus, `with_glib` option is deprecated, and now removed.

#### Details

The upstream defines specific paths when not configured: https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.15.8/meson.build?ref_type=tags#L956

It will be better only passing values when the option is configured.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by garethsb — APPROVED** [review]
LGTM, thanks!

**Review by RubenRBS — APPROVED** [review]
Good catch :)

File: recipes/dbus/1.x.x/conanfile.py [hunk 1]
```diff
@@ -134,8 +130,10 @@ def generate(self):
             tc.project_options["message_bus"] = bool(self.options.message_bus)
         if self.options.get_safe("dbus_user"):
             tc.project_options["dbus_user"] = str(self.options.dbus_user)
-        tc.project_options["system_pid_file"] = str(self.options.get_safe("system_pid_file", ""))
-        tc.project_options["system_socket"] = str(self.options.get_safe("system_socket", ""))
+        if self.options.system_pid_file:
```
Comment (garethsb) [inline recipes/dbus/1.x.x/conanfile.py:133]: @uilianries I think maybe these ifs should all use `.get_safe()` to handle the case that the key is not present in options at all, or can that never happen?

Comment (RubenRBS) [reply recipes/dbus/1.x.x/conanfile.py:133]: That can never happen, as it's not being deleted in the config_options/configure methods :)

File: recipes/dbus/1.x.x/conanfile.py [hunk 2]
```diff
@@ -134,8 +130,10 @@ def generate(self):
...
+            tc.project_options["system_pid_file"] = self.options.system_pid_file
```
Comment (franramirez688) [inline recipes/dbus/1.x.x/conanfile.py:134]:
```suggestion
            tc.project_options["system_pid_file"] = str(self.options.system_pid_file)
```

File: recipes/dbus/1.x.x/conanfile.py [hunk 3]
```diff
@@ -134,8 +130,10 @@ def generate(self):
...
+        if self.options.system_socket:
+            tc.project_options["system_socket"] = self.options.system_socket
```
Comment (franramirez688) [inline recipes/dbus/1.x.x/conanfile.py:136]:
```suggestion
            tc.project_options["system_socket"] = str(self.options.system_socket)
```

**Review by franramirez688 — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**